codetf.schema.json

```diff
@@ -0,0 +1,228 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://pixee.ai/codetf.schema.json",
+  "title": "Code Transformation Format (CodeTF)",
+  "description": "A format for representing the results of code transformations",
+  "type": "object",
+
+  "properties": {
+    "run": {
+      "description": "Metadata about the run that produced the results",
+      "$ref": "#/definitions/run"
+    },
+    "results": {
+      "description": "The results of the code transformations that were applied during analysis",
+      "type": "array",
+      "items": { "$ref": "#/definitions/result" },
+      "minItems": 0
+    }
+  },
+
+  "definitions": {
+    "run": {
+      "type": "object",
+      "properties": {
+        "vendor": {
+          "type": "string",
+          "description": "Name of the organization that produced the tool"
+        },
+        "tool": {
+          "type": "string",
+          "description": "Name of the tool that produced the CodeTF results"
+        },
+        "version": {
+          "type": "string",
+          "description": "Version of the tool that produced the CodeTF results"
+        },
+        "projectName": {
+          "type": "string",
+          "description": "Name of the project that was analyzed"
+        },
+        "commandLine": {
+          "type": "string",
+          "description": "The command line used to re-create this run"
+        },
+        "elapsed": {
+          "type": ["integer", "string"],
+          "description": "How long, in miliseconds, the analysis took to execute"
+        },
+        "directory": {
+          "type": "string",
+          "description": "The location on disk of the code directory being scanned"
+        },
+        "sarifs": {
+          "type": "array",
+          "items": { "$ref": "#/definitions/sarif" }
+        }
+      },
+      "required": ["vendor", "tool", "version", "commandLine", "elapsed", "directory"]
+
+    },
+
+    "result": {
+      "type": "object",
+      "properties": {
+        "codemod": {
+          "type": "string",
+          "description": "ID of the transformation that was applied"
+        },
+        "summary": {
+          "type": "string",
+          "description": "A short phrase describing the changes made"
+        },
+        "description": {
+          "type": "string",
+          "description": "A longer description of the changes made"
+        },
+        "detectionTool": {
```

Comment:
The debate for this field properly belongs in #19. I'm happy to remove it here if we just want to move ahead with this change.